lib/util/git/private-key.ts

```diff
@@ -60,11 +71,60 @@ class GPGKey extends PrivateKey {
   }
 }
 
+class SSHKey extends PrivateKey {
+  protected readonly gpgFormat = 'ssh';
+
+  protected async importKey(): Promise<string | undefined> {
+    const keyFileName = upath.join(os.tmpdir() + '/git-private-ssh.key');
+    if (await this.hasPassphrase(keyFileName)) {
+      throw new Error('SSH key must have an empty passhprase');
+    }
+    await fs.outputFile(keyFileName, this.key);
+    await fs.chmod(keyFileName, 0o600);
+    // HACK: `git` calls `ssh-keygen -Y sign ...` internally for SSH-based
+    // commit signing. Technically, only the private key is needed for signing,
+    // but `ssh-keygen` has an implementation quirk which requires also the
+    // public key file to exist. Therefore, we derive the public key from the
+    // private key just to satisfy `ssh-keygen` until the problem has been
+    // resolved.
+    // https://github.com/renovatebot/renovate/issues/18197#issuecomment-2152333710
+    const { stdout } = await exec(`ssh-keygen -y -P "" -f ${keyFileName}`);
+    await fs.outputFile(`${keyFileName}.pub`, stdout);
```

Comment:
private key file needs to be removed when renovate exits.

Reply:
Good point! :ok_hand: I've added exit handlers to delete both private and the public key files and amended a test to check for their successful removal: 9ee2ebca2f42e17fb0ed4345c2185c16df5d6ed8